tracboat/gitlab/__init__.py

```diff
@@ -1 +1,93 @@
 # -*- coding: utf-8 -*-
+
+import abc
+
+import six
+
+
+__all__ = ['ConnectionBase', 'split_project_components']
+
+
+def split_project_components(project_name):
+    if '/' in project_name:
+        return project_name.split('/')
+    else:
+        return None, project_name
+
+
+@six.add_metaclass(abc.ABCMeta)
+class ConnectionBase(object):
+
+    def __init__(self, project_name):
+        project = project_name.strip()
+        if not project:
+            raise ValueError('invalid project name: {!r}'.format(project_name))
+        namespace, project = split_project_components(project)
```

Comment:
Thanks, I was not aware of nested groups, I have to figure out how are they stored in the database schema, I'm going to try with GitLab 9.0.0...